draft-irtf-cfrg-opaque.md

```diff
@@ -1673,6 +1673,22 @@ applications can use to control OPAQUE:
   the oprf_seed) for all clients; see {{preventing-client-enumeration}}. In settings
   where this attack is not a concern, servers may choose to not support this functionality.
 
+# Implementation Considerations {#implementation-considerations}
+
+Implementations of OPAQUE should consider addressing the following:
+
+- Clearing secrets out of memory: All private key material and intermediate values,
+including the outputs of the key exchange phase, should not be retained in memory after
+deallocation.
+- Constant-time operations: All operations, particularly the cryptographic and group
+arithmetic operations, should be constant-time and independent of the bits of any secrets.
+This includes any conditional branching during the creation of the credential response,
+to support implementations which provide mitigations against client enumeration attacks.
+- Deserialization checks: When parsing messages that have crossed trust boundaries (e.g.
```

Comment:
Can we just cite the VOPRF draft for this? It's unclear what "element" means in this context, and it's no longer the responsibility of OPAQUE to do this check since VOPRF handles it accordingly. Instead, we might say that implementations must handle errors raised by the VOPRF APIs and abort accordingly. 

Thoughts?

Reply:
Sounds good, the new text reads:

```
- Deserialization checks: When parsing messages that have crossed trust boundaries (e.g.
a network wire), implementations should properly handle all error conditions covered in
{{I-D.irtf-cfrg-voprf}} and abort accordingly.
```